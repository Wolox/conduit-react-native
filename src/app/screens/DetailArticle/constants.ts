import { darkGray, green } from '@constants/colors';
import { SPACING } from '@constants/dimentions';
import { UserResponse } from '@interfaces/authInterfaces';
import { MixedStyleDeclaration } from 'react-native-render-html';

export function MOCK_DATA({ user: { username } }: UserResponse, comment: string) {
  return {
    author: {
      bio: 'Bio de prebas',
      following: false,
      image:
        'https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8dXNlcnxlbnwwfHwwfHw%3D&ixlib=rb-1.2.1&w=1000&q=80',
      username
    },
    body: comment,
    createdAt: '2021-08-06T18:16:15.699Z',
    updatedAt: '2021-08-06T18:16:15.699Z',
    id: Number(Math.random().toFixed(2))
  };
}
enum COMMON_STYLES {
  COMMON_TITLE_FONT = 14,
  PARAGRAPH_SIZE = 12,
  NONE = 'none',
  NORMAL = 'normal'
}
const TITLE_STYLES: MixedStyleDeclaration = {
  fontSize: COMMON_STYLES.COMMON_TITLE_FONT,
  padding: SPACING.NONE,
  marginVertical: SPACING.XXSMALL
};
export const tagsStyles: Readonly<Record<string, MixedStyleDeclaration>> = {
  body: {
    whiteSpace: COMMON_STYLES.NORMAL,
    color: darkGray
  },
  h1: {
    ...TITLE_STYLES
  },
  h2: {
    ...TITLE_STYLES
  },
  p: {
    fontSize: COMMON_STYLES.PARAGRAPH_SIZE,
    padding: SPACING.NONE,
    marginVertical: SPACING.XXSMALL
  },
  a: {
    color: green,
    textDecorationLine: COMMON_STYLES.NONE
  },
  ol: {
    margin: SPACING.NONE,
    padding: SPACING.NONE
  },
  li: {
    paddingBottom: SPACING.XXSMALL
  }
};
