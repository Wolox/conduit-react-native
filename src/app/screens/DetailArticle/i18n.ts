import i18next from 'i18next';

i18next.addResources('en', 'DETAIL_ARTICLE', {
  NO_COMMENTS_TEXT: ' No comments yet, Do the first!',
  PLACE_HOLDER_INPUT: 'Insert your comment...',
  NO_AUTHENTICATED: 'Register to interact with other users',
  MIN_LENGTH_COMMENT: 'A comment requires at least 5 characters'
});
