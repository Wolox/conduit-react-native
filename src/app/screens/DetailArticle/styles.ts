import { StyleSheet } from 'react-native';
import { darkGray, lightGray, white, red, gray, green } from '@constants/colors';
import { SPACING, BORDER_WIDTH } from '@constants/dimentions';
import { BOTTOM_SEPARATOR_LINE, commonBoxShadow } from '@constants/commonStyles';
import { moderateScale } from '@utils/scalingUtils';
import fonts from '@config/fonts';

const IMAGE_USER_SIZE = moderateScale(80);
const INTERACION_ICON_SIZE = moderateScale(30);

export default StyleSheet.create({
  container: {
    flex: 1,
    position: 'relative',
    alignItems: 'center',
    padding: SPACING.MEDIUM,
    paddingTop: SPACING.XXBIG * SPACING.XXXSMALL
  },
  containerActions: {
    width: '100%',
    height: INTERACION_ICON_SIZE,
    position: 'absolute',
    top: SPACING.MEDIUM,
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  containerDetail: {
    ...commonBoxShadow,
    position: 'relative',
    padding: SPACING.MEDIUM,
    borderRadius: SPACING.MEDIUM,
    backgroundColor: white,
    paddingTop: SPACING.XXBIG
  },
  containerUser: {
    marginBottom: SPACING.MEDIUM,
    justifyContent: 'center',
    alignItems: 'center'
  },
  image: {
    position: 'absolute',
    top: -SPACING.XXBIG * SPACING.XXXSMALL,
    width: IMAGE_USER_SIZE,
    height: IMAGE_USER_SIZE,
    backgroundColor: darkGray,
    borderRadius: SPACING.XXBIG
  },
  interactionButtons: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    marginTop: SPACING.SMALL,
    height: INTERACION_ICON_SIZE,
    width: '100%'
  },
  interactionButton: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    marginHorizontal: SPACING.XSMALL
  },
  interactionButtonImage: {
    width: INTERACION_ICON_SIZE,
    height: INTERACION_ICON_SIZE,
    marginRight: SPACING.XSMALL,
    tintColor: gray
  },
  separator: {
    ...BOTTOM_SEPARATOR_LINE,
    marginVertical: SPACING.SMALL
  },
  tagContainer: {
    flexDirection: 'row'
  },
  tag: {
    ...fonts.labelFont,
    borderColor: lightGray,
    borderWidth: BORDER_WIDTH.BOLD,
    padding: SPACING.XXXSMALL,
    marginHorizontal: SPACING.XXXSMALL
  },
  redTint: {
    tintColor: red
  },
  greenTint: {
    tintColor: green
  }
});
