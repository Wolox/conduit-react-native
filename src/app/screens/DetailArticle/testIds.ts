export default {
  favoriteButton: 'favorite-button',
  followButton: 'follow-button'
};
