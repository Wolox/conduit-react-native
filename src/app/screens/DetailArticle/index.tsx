import React, { useState, useCallback } from 'react';
import { View, Image, TouchableOpacity, Text } from 'react-native';
import { useSelector } from 'react-redux';
import CustomText from '@components/CustomText';
import useNavigation from '@components/AppNavigator/helper';
import { State } from '@interfaces/reduxInterfaces';
import { ArticleInParams } from '@interfaces/articlesInterface';
import { formatDate } from '@utils/dateUtils';
import icAddInactive from '@assets/TabBar/icAddpostInactive.png';
import icAddActive from '@assets/TabBar/icAddpostActive.png';
import icFavouriteInactive from '@assets/TabBar/icFavoriteInactive.png';
import icFavouriteActive from '@assets/TabBar/icFavoriteActive.png';
import icDefaultArticleImage from '@assets/icons/icDefaultArticleImage.jpg';
import icDelete from '@assets/icons/icDelete.png';
import icEdit from '@assets/icons/icEdit.png';
import Routes from '@constants/routes';

import styles from './styles';
import testIds from './testIds';

interface Props extends ArticleInParams {}

function DetailArticle({ route }: Props) {
  const {
    title,
    description,
    updatedAt,
    favoritesCount,
    author: { image, username, following },
    tagList
  } = route?.params?.article;
  const currentUser = useSelector((state: State) => state.auth.currentUser);
  const navigation = useNavigation();
  const [favoriteCount, setFavoriteCount] = useState(favoritesCount || 0);
  const [isFollow, setIsFollow] = useState(following);
  const handleToggleFavorite = () => {
    if (favoriteCount > favoritesCount) setFavoriteCount(favoriteCount - 1);
    else setFavoriteCount(favoriteCount + 1);
  };
  const handleDeleteArticle = () => console.log('delete article');

  const handleEditArticle = useCallback(
    () =>
      navigation?.navigate(Routes.EditArticle, {
        article: { ...route?.params?.article, isEditArticle: true }
      }),
    [navigation, route]
  );

  return (
    <View style={styles.container}>
      {currentUser && (
        <View style={styles.containerActions}>
          <TouchableOpacity
            testID={testIds.editButton}
            style={styles.interactionButton}
            onPress={handleEditArticle}>
            <Image
              style={[styles.interactionButtonImage, styles.greenTint]}
              source={icEdit}
              resizeMode="contain"
            />
          </TouchableOpacity>
          <TouchableOpacity
            testID={testIds.deleteButton}
            style={styles.interactionButton}
            onPress={handleDeleteArticle}>
            <Image
              style={[styles.interactionButtonImage, styles.redTint]}
              source={icDelete}
              resizeMode="contain"
            />
          </TouchableOpacity>
        </View>
      )}
      <View style={styles.containerDetail}>
        <View style={styles.containerUser}>
          <Image source={image ? { uri: image } : icDefaultArticleImage} style={styles.image} />
          <View>
            <CustomText center green>
              {username}
            </CustomText>
            <CustomText center label>
              {formatDate(updatedAt)}
            </CustomText>
          </View>
        </View>
        {!!tagList.length && (
          <View style={styles.tagContainer}>
            {tagList.map((tag: string, index: number) => (
              <Text style={styles.tag} key={`key-${tag}-${index}`}>
                {tag}
              </Text>
            ))}
          </View>
        )}
        <View style={styles.separator} />
        <CustomText>{title}</CustomText>
        <CustomText label>{description}</CustomText>
        <View style={styles.interactionButtons}>
          <TouchableOpacity
            testID={testIds.followButton}
            style={styles.interactionButton}
            onPress={() => setIsFollow(!isFollow)}>
            <Image
              style={styles.interactionButtonImage}
              source={isFollow ? icAddActive : icAddInactive}
              resizeMode="contain"
            />
          </TouchableOpacity>
          <TouchableOpacity
            testID={testIds.favoriteButton}
            style={styles.interactionButton}
            onPress={handleToggleFavorite}>
            <Image
              style={styles.interactionButtonImage}
              source={favoriteCount > favoritesCount ? icFavouriteActive : icFavouriteInactive}
              resizeMode="contain"
            />
            {!!favoriteCount && (
              <CustomText
                gray
                xsmall
                green={favoriteCount > favoritesCount}>{`(${favoriteCount})`}</CustomText>
            )}
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
}

export default DetailArticle;
