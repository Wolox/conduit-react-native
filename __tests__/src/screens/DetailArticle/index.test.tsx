import React from 'react';
import renderer from 'react-test-renderer';
import DetailArticle from '@screens/DetailArticle';
import userIcon from '@assets/Profile/icUser.png';
import MockedNavigator from '@mocks/MockedNavigator';

jest.mock('react-redux', () => {
  return {
    useSelector: jest.fn().mockImplementation(() => ({ comments: [], currentUser: '' })),
    useDispatch: () => jest.fn()
  };
});

describe('DetailArticle', () => {
  test('DetailArticle Snapshot', () => {
    const item = {
      route: {
        params: {
          article: {
            title: '',
            description: '',
            updatedAt: '',
            body: '',
            favoritesCount: 0,
            slug: '',
            author: { image: userIcon, username: 'user', following: false },
            tagList: []
          }
        }
      }
    };

    const RenderCustom = {
      renderDetail: () => <DetailArticle {...item} />,
      renderCompleteDetail: () => <MockedNavigator component={RenderCustom.renderDetail} />
    };
    const tree = renderer.create(RenderCustom.renderCompleteDetail()).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
