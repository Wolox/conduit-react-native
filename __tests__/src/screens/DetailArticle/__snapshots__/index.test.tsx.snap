// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailArticle DetailArticle Snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                          "padding": 16,
                          "paddingTop": 96,
                          "position": "relative",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "height": 39.285714285714285,
                            "justifyContent": "space-between",
                            "position": "absolute",
                            "top": 16,
                            "width": "100%",
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "marginHorizontal": 6,
                              "opacity": 1,
                            }
                          }
                          testID="edit-button"
                        >
                          <Image
                            resizeMode="contain"
                            source={
                              Object {
                                "process": [Function],
                              }
                            }
                            style={
                              Array [
                                Object {
                                  "height": 39.285714285714285,
                                  "width": 39.285714285714285,
                                },
                                Object {
                                  "tintColor": "#5CB85C",
                                },
                              ]
                            }
                          />
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "marginHorizontal": 6,
                              "opacity": 1,
                            }
                          }
                          testID="delete-button"
                        >
                          <Image
                            resizeMode="contain"
                            source={
                              Object {
                                "process": [Function],
                              }
                            }
                            style={
                              Array [
                                Object {
                                  "height": 24,
                                  "marginRight": 4,
                                  "width": 24,
                                },
                                Object {
                                  "tintColor": "#FF0000",
                                },
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "backgroundColor": "#FFF",
                            "borderRadius": 16,
                            "elevation": 1,
                            "padding": 16,
                            "paddingTop": 48,
                            "position": "relative",
                            "shadowColor": "#E5E5E5",
                            "shadowOffset": Object {
                              "height": 3,
                              "width": 0,
                            },
                            "shadowOpacity": 1,
                            "width": "100%",
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "justifyContent": "center",
                                "marginBottom": 16,
                              }
                            }
                          >
                            <Image
                              source=""
                              style={
                                Object {
                                  "backgroundColor": "#F1F1F1",
                                  "borderRadius": 48,
                                  "height": 125.71428571428571,
                                  "position": "absolute",
                                  "top": -96,
                                  "width": 125.71428571428571,
                                }
                              }
                            />
                            <View>
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "backgroundColor": "transparent",
                                      "color": "#000",
                                      "fontFamily": "Nunito",
                                      "fontSize": 25.142857142857142,
                                      "fontStyle": "normal",
                                      "fontWeight": "400",
                                    },
                                    Array [
                                      Object {
                                        "textAlign": "center",
                                      },
                                      Object {
                                        "color": "#5CB85C",
                                      },
                                    ],
                                    undefined,
                                  ]
                                }
                              />
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "backgroundColor": "transparent",
                                      "color": "#000",
                                      "fontFamily": "Nunito",
                                      "fontSize": 25.142857142857142,
                                      "fontStyle": "normal",
                                      "fontWeight": "400",
                                    },
                                    Array [
                                      Object {
                                        "textAlign": "center",
                                      },
                                      Object {
                                        "color": "#939393",
                                        "fontFamily": "Nunito",
                                        "fontSize": 18.857142857142858,
                                        "fontStyle": "normal",
                                        "fontWeight": "400",
                                      },
                                    ],
                                    undefined,
                                  ]
                                }
                              >
                                06 · 12 · 2021
                              </Text>
                            </View>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "borderBottomColor": "#E5E5E5",
                              "borderBottomWidth": 1,
                              "marginVertical": 12,
                            }
                          }
                        />
                        <Text
                          style={
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                                "color": "#000",
                                "fontFamily": "Nunito",
                                "fontSize": 25.142857142857142,
                                "fontStyle": "normal",
                                "fontWeight": "400",
                              },
                              Array [],
                              undefined,
                            ]
                          }
                        />
                        <Text
                          style={
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                                "color": "#000",
                                "fontFamily": "Nunito",
                                "fontSize": 25.142857142857142,
                                "fontStyle": "normal",
                                "fontWeight": "400",
                              },
                              Array [
                                Object {
                                  "color": "#939393",
                                  "fontFamily": "Nunito",
                                  "fontSize": 18.857142857142858,
                                  "fontStyle": "normal",
                                  "fontWeight": "400",
                                },
                              ],
                              undefined,
                            ]
                          }
                        />
                        <View
                          style={
                            Object {
                              "marginBottom": 20,
                              "marginTop": 15,
                              "minWidth": "90%",
                            }
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "color": "#000",
                                  "fontFamily": "Nunito",
                                  "fontSize": 25.142857142857142,
                                  "fontStyle": "normal",
                                  "fontWeight": "400",
                                },
                                Array [
                                  Object {
                                    "color": "#939393",
                                    "fontFamily": "Nunito",
                                    "fontSize": 18.857142857142858,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                  },
                                ],
                                undefined,
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                              "height": 39.285714285714285,
                              "justifyContent": "flex-end",
                              "marginTop": 12,
                              "width": "100%",
                            }
                          }
                        >
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "justifyContent": "center",
                                "marginHorizontal": 6,
                                "opacity": 1,
                              }
                            }
                            testID="follow-button"
                          >
                            <Image
                              resizeMode="contain"
                              source={
                                Object {
                                  "process": [Function],
                                }
                              }
                              style={
                                Array [
                                  Object {
                                    "height": 39.285714285714285,
                                    "marginRight": 6,
                                    "tintColor": "#A7A7A7",
                                    "width": 39.285714285714285,
                                  },
                                  undefined,
                                ]
                              }
                            />
                          </View>
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "justifyContent": "center",
                                "marginHorizontal": 6,
                                "opacity": 1,
                              }
                            }
                            testID="favorite-button"
                          >
                            <Image
                              resizeMode="contain"
                              source={
                                Object {
                                  "process": [Function],
                                }
                              }
                              style={
                                Array [
                                  Object {
                                    "height": 39.285714285714285,
                                    "marginRight": 6,
                                    "tintColor": "#A7A7A7",
                                    "width": 39.285714285714285,
                                  },
                                  Object {
                                    "height": 39.285714285714285,
                                    "marginRight": 6,
                                    "tintColor": "#A7A7A7",
                                    "width": 39.285714285714285,
                                  },
                                ]
                              }
                            />
                          </View>
                        </View>
                      </View>
                    </View>
                    <RCTScrollView>
                      <View />
                    </RCTScrollView>
                    <View
                      style={
                        Object {
                          "marginHorizontal": 2,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "marginTop": 4,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "marginBottom": 5,
                                "padding": 10,
                              },
                              false,
                              Object {
                                "marginBottom": 0,
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#FFF",
                                  "borderRadius": 15,
                                  "elevation": 1,
                                  "height": 75,
                                  "paddingHorizontal": 5,
                                  "shadowColor": "#E5E5E5",
                                  "shadowOffset": Object {
                                    "height": 3,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 1,
                                },
                                Object {},
                                Array [
                                  Object {
                                    "height": 300,
                                    "paddingBottom": 10,
                                  },
                                  Object {
                                    "height": 125,
                                  },
                                ],
                                Object {
                                  "flexDirection": "row",
                                },
                              ]
                            }
                          >
                            <TextInput
                              allowFontScaling={false}
                              autoCapitalize="none"
                              autoCompleteType="off"
                              autoCorrect={false}
                              blurOnSubmit={true}
                              clearButtonMode="never"
                              editable={true}
                              keyboardType="default"
                              maxLength={300}
                              multiline={true}
                              name="message"
                              numberOfLines={10}
                              onBlur={[Function]}
                              onChangeText={[Function]}
                              onFocus={[Function]}
                              onKeyPress={[Function]}
                              placeholder="DETAIL_ARTICLE:PLACE_HOLDER_INPUT"
                              rejectResponderTermination={true}
                              returnKeyType="done"
                              style={
                                Array [
                                  Object {
                                    "color": "#555555",
                                    "fontFamily": "Nunito",
                                    "fontSize": 14,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                    "margin": 0,
                                    "padding": 0,
                                  },
                                  false,
                                  Object {
                                    "color": "#000",
                                    "fontSize": 14,
                                    "marginBottom": 20,
                                    "paddingHorizontal": 8,
                                    "paddingTop": 8,
                                    "paddingVertical": 0,
                                    "textAlignVertical": "top",
                                  },
                                  Object {
                                    "width": "80%",
                                  },
                                ]
                              }
                              underlineColorAndroid="transparent"
                              value=""
                            />
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "height": "100%",
                                  "justifyContent": "center",
                                  "width": "20%",
                                }
                              }
                            >
                              <View
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Image
                                  resizeMethod="scale"
                                  resizeMode="contain"
                                  source={
                                    Object {
                                      "process": [Function],
                                    }
                                  }
                                  style={
                                    Array [
                                      Object {
                                        "height": 23,
                                        "tintColor": "#5CB85C",
                                        "width": 23,
                                      },
                                    ]
                                  }
                                />
                              </View>
                            </View>
                            <Text
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "color": "#000",
                                    "fontFamily": "Nunito",
                                    "fontSize": 25.142857142857142,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                  },
                                  Array [
                                    Object {
                                      "color": "#A7A7A7",
                                    },
                                    Object {
                                      "fontSize": 18.857142857142858,
                                    },
                                  ],
                                  Array [
                                    Object {
                                      "color": "#E5E5E5",
                                      "marginHorizontal": 5,
                                    },
                                    Object {
                                      "bottom": 0,
                                      "margin": 10,
                                      "position": "absolute",
                                      "right": 0,
                                    },
                                  ],
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "backgroundColor": "transparent",
                                      "color": "#000",
                                      "fontFamily": "Nunito",
                                      "fontSize": 25.142857142857142,
                                      "fontStyle": "normal",
                                      "fontWeight": "400",
                                    },
                                    Array [
                                      Object {
                                        "color": "#A7A7A7",
                                      },
                                    ],
                                    undefined,
                                  ]
                                }
                              >
                                0
                              </Text>
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "backgroundColor": "transparent",
                                      "color": "#000",
                                      "fontFamily": "Nunito",
                                      "fontSize": 25.142857142857142,
                                      "fontStyle": "normal",
                                      "fontWeight": "400",
                                    },
                                    Array [
                                      Object {
                                        "color": "#A7A7A7",
                                      },
                                    ],
                                    undefined,
                                  ]
                                }
                              >
                                 / 300
                              </Text>
                            </Text>
                          </View>
                          <View
                            style={
                              Array [
                                Object {
                                  "height": 15,
                                  "marginTop": 3,
                                },
                                undefined,
                              ]
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
  </RNCSafeAreaProvider>
</View>
`;
